src/ORM/Association/BelongsToMany.php

```diff
@@ -586,9 +586,16 @@ public function eagerLoader(array $options)
      */
     public function cascadeDelete(EntityInterface $entity, array $options = [])
     {
-        if (!$this->getDependent()) {
+        $dependent = $this->getDependent();
```

Comment:
Doesn't this change the return types of `getDependent()` and `setDependent()` as new types can be provided/fetched?